pow/src/main/java/tech/pegasys/teku/pow/exception/Eth1RequestExceptionsContainer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.pow.exception;
+
+import com.google.common.base.Throwables;
+import java.net.SocketTimeoutException;
+import java.util.ArrayList;
+import java.util.Optional;
+
+public class Eth1RequestExceptionsContainer extends RuntimeException {
+  private Optional<ArrayList<Throwable>> thrownErrors;
+
+  public Eth1RequestExceptionsContainer() {
+    super("One or more eth1 endpoints thrown an Exception");
+    thrownErrors = Optional.empty();
+  }
+
+  public void add(Throwable error) {
+    // Lazily init ArrayList
+    thrownErrors =
+        thrownErrors
+            .or(() -> Optional.of(new ArrayList<>()))
+            .map(
+                al -> {
+                  al.add(error);
+                  return al;
+                });
+  }
```

Comment:
This pattern is slightly out of character for an exception but I like the general idea.  I think what you want to use is the inherited `Throwable.addSuppressed(exception)` rather than this custom `add`.  You can then get them with `getSuppressed`.  That has the advantage that when you print the exception stack trace, all the suppressed exceptions are included so we'd wind up with a record of why each endpoint failed.

Reply:
Ah! wasn't aware of add\getSuppressed, that's perfect here.